network/network_windows.go

```diff
@@ -112,15 +115,169 @@ func (nm *networkManager) newNetworkImpl(nwInfo *NetworkInfo, extIf *externalInt
 	return nw, nil
 }
 
+// newNetworkImplHnsV2 creates a new container network for HNSv2.
+func (nm *networkManager) newNetworkImplHnsV2(nwInfo *NetworkInfo, extIf *externalInterface) (*network, error) {
+	// Initialize HNS network.
+	hnsNetwork := &hcn.HostComputeNetwork{
+		Name: nwInfo.Id,
+		Dns: hcn.Dns{
+			Domain:     nwInfo.DNS.Suffix,
+			ServerList: nwInfo.DNS.Servers,
+		},
+		Ipams: []hcn.Ipam{
+			hcn.Ipam{
+				Type: "Static",
+			},
+		},
+		SchemaVersion: hcn.SchemaVersion{
+			Major: 2,
+			Minor: 0,
+		},
+	}
+
+	// Set hcn network adaptor name policy
+	// FixMe: Find a better way to check if a nic that is selected is not part of a vSwitch
+	if !strings.HasPrefix(extIf.Name, "vEthernet") {
+		netAdapterNamePolicy, err := policy.GetHcnNetAdapterPolicy(extIf.Name)
+		if err != nil {
+			log.Printf("[net] Failed to serialize network adapter policy due to error: %v", err)
+			return nil, err
+		}
+
+		hnsNetwork.Policies = append(hnsNetwork.Policies, netAdapterNamePolicy)
+	}
+
+	// Set hcn subnet policy
+	var vlanid int
+	var subnetPolicy []byte
+	opt, _ := nwInfo.Options[genericData].(map[string]interface{})
+	if opt != nil && opt[VlanIDKey] != nil {
+		var err error
+		vlanID, _ := strconv.ParseUint(opt[VlanIDKey].(string), 10, 32)
+		subnetPolicy, err = policy.SerializeHcnSubnetVlanPolicy((uint32)(vlanID))
+		if err != nil {
+			log.Printf("[net] Failed to serialize subnet vlan policy due to error: %v", err)
+			return nil, err
+		}
+
+		vlanid = (int)(vlanID)
+	}
+
+	// Set network mode.
+	switch nwInfo.Mode {
+	case opModeBridge:
+		hnsNetwork.Type = hcn.L2Bridge
+	case opModeTunnel:
```

Comment:
seems like we already have this logic, as Jareyn mentioned we could encapsulate lots of this logic in place.